solver/src/oneinch_solver.rs

```diff
@@ -92,6 +101,20 @@ impl OneInchSolver {
     fn web3(&self) -> DynWeb3 {
         self.settlement_contract.raw_instance().web3()
     }
+
+    async fn chain_id(&self) -> Result<u64> {
```

Comment:
We already load the chain_id on startup [here](https://github.com/gnosis/oba-services/blob/bed59910aedace2e04fd28e7d04e1ed5a5114f78/solver/src/main.rs#L132). Can we not pass it in from above?

Reply:
That makes a lot of sense.